servers/cu/src/routes/middleware/withProcessRestriction.js

```diff
@@ -0,0 +1,12 @@
+export const withProcessRestriction = (handler) => (req, res, next) => {
+  const {
+    params: { processId },
```

Comment:
Not all endpoints will have `processId` as a resource in the path ie. `.../:processId`. Some endpoints, like `/dry-run`, receive `process-id` as a query parameter.

Here's a exhaustive list of CU endpoints and where to retrieve the `processId`, for each

- `/cron/:processId` (path)
- `/dry-run?process-id=...` (query param)
- `/result/:messageId?process-id=...` (query param)
- `/results/:processId` (path)
- `/state/:processId` (path)

So you could consider creating a middleware that can dynamically extract the `processId` from the `req`, say a HoF, using an "`extractor`" injected as a parameter:

```js
const withProcessRestrictionFrom = ({ extractor }) => (handler) => (req, res, next) => {
  const processId = extractor(req)

  if (...) return res.status(403).send(...)

  return handler(req, res, next)
}

const withProcessRestrictionFromPath = withProcessRestrictionFrom({ extractor: (req) => req.params.processId })
const withProcessRestrictionFromQuery = withProcessRestrictionFrom({ extractor: (req) => req.query['process-id'] })
```

Then you could simply compose the appropriate middleware, depending on the route, instead of as part of `withMiddleware`

I hope that description makes sense. Other valid approaches as well.